src/DataWrangling/masking.jl

```diff
@@ -157,3 +228,45 @@ function inpaint_mask!(field, mask; inpainting=NearestNeighborInpainting(Inf))
     return field
 end
 
+function default_set_dataset_mask end
+
+"""
+    dataset_mask(metadata, architecture = CPU();
+                 data_field = Field(metadata; architecture, inpainting=nothing),
+                 minimum_value = Float32(-1e5),
+                 maximum_value = Float32(1e5))
+
+A boolean field where `true` represents a missing value in the dataset.
+"""
+function dataset_mask(metadata, architecture = CPU();
+                      data_field = Field(metadata; architecture, inpainting=nothing),
+                      minimum_value = Float32(-1e5),
+                      maximum_value = Float32(1e5))
+
+    mask  = Field{location(data_field)...}(data_field.grid, Bool)
+
+    _set_mask! = default_set_dataset_mask(metadata)
+
+    # Set the mask with zeros where field is defined
+    launch!(architecture, data_field.grid, :xyz, _set_mask!, mask, data_field, minimum_value, maximum_value)
+
+    return mask
+end
+
+"""
+    dataset_immersed_grid(metadata, architecture = CPU())
+
+Compute the `ImmersedBoundaryGrid` for `metadata` with a bottom height field that is defined
+by the first non-missing value from the bottom up.
+"""
+function dataset_immersed_grid(metadata, architecture = CPU())
+
+    mask = dataset_mask(metadata, architecture)
+    grid = mask.grid
+    bottom = Field{Center, Center, Nothing}(grid)
+
+    # Set the mask with zeros where field is defined
+    launch!(architecture, grid, :xy, _set_height_from_mask!, bottom, grid, mask)
+
+    return ImmersedBoundaryGrid(grid, GridFittedBottom(bottom))
```

Comment:
Will this work? the mask is 3D while `bottom` is 2D. Probably we need to just
```suggestion
    return ImmersedBoundaryGrid(grid, GridFittedBoundary(mask))
```

Reply:
This is on `main`:

https://github.com/CliMA/ClimaOcean.jl/blob/46487980688544cdffe88acc86d47ce42f162909/src/DataWrangling/ECCO/ECCO_mask.jl#L59

So let's leave this for a different PR. I think there is an example that uses `ECCO_immersed_grid()` but the docs are broken atm.

Reply:
Oh I haven't seen the `_set_height_from_mask!` kernel, no this would definitely work!